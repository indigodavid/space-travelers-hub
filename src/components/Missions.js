import React from 'react';
import { useSelector } from 'react-redux';
import JoinMission from './JoinMission';
import MissionBadge from './MissionBadge';

const Missions = () => {
  const missions = useSelector((state) => state.missions);
  return (
    <ul className="missionsData">
      {missions.map((mission) => (
        <li key={mission.mission_id}>
          <div className="missionName">{mission.mission_name}</div>
          <div className="description">{mission.description}</div>
          <div className="status">
            <MissionBadge reserved={mission.reserved} />
            <JoinMission reserved={mission.reserved} id={mission.mission_id} />
          </div>
        </li>
      ))}
    </ul>
  );
};

export default Missions;
