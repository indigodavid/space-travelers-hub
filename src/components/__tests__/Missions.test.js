import React from 'react';
import { render, screen } from '@testing-library/react';
import '@testing-library/jest-dom';
import { Provider } from 'react-redux';
import store from '../../redux/configureStore'
import Missions from '../Missions';


const renderMissionsComponent = () => {
  const { container } = render(
    <Provider store={store}>
      <Missions />
    </Provider>
  );
  return container;
}

describe('Test Missions component', () => {
  test('Match Snapshot', () => {
    expect(renderMissionsComponent().firstChild).toMatchSnapshot();
  })

  test('Find Elements ny text', () => {
    expect(renderMissionsComponent().querySelector('.missionsData')).toBeInTheDocument();
  })
})

