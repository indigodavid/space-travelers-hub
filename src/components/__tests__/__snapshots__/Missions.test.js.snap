// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test Missions component Match Snapshot 1`] = `
<ul
  class="missionsData"
/>
`;
